### ISSUE TITLE
Set-Service Credential Parameter Missing

### ISSUE THREAD

**Issue by chrisecklar** [PowerShell/vscode-powershell#4849]
### Prerequisites

- [X] I have written a descriptive issue title.
- [X] I have searched all [_open and closed_ issues](https://github.com/PowerShell/vscode-powershell/issues?q=is%3Aissue) to ensure it has not already been reported.
- [X] I have read the [troubleshooting](https://github.com/PowerShell/vscode-powershell/blob/main/docs/troubleshooting.md) guide.
- [X] I am sure this issue is with the _extension itself_ and does not reproduce in a standalone [PowerShell](https://github.com/PowerShell/PowerShell/issues/new/choose) instance.
- [X] I have verified that I am using the latest version of Visual Studio Code and the PowerShell extension.
- [X] If this is a security issue, I have read the [security issue reporting guidance](https://github.com/PowerShell/vscode-powershell/blob/main/SECURITY.md).

### Summary

I cannot use Set-Service -Name "Service" -Credential as the Credential parameter appears to have been recently removed.  I am running **Pwsh 7.4.0** **VS Code 1.84.2** and the **PowerShell extension 2023.08.0**.  **_The issue does not occur when using the Pwsh console directly--only inside of VS Code._**

The following Microsoft Learn documentation states that this parameter should still be available:

https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.management/set-service?view=powershell-7.4

### PowerShell Version

```console
Name                           Value
----                           -----
PSVersion                      7.4.0
PSEdition                      Core
GitCommitId                    7.4.0
OS                             Microsoft Windows 10.0.22621
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visual Studio Code Version

```console
1.84.2
1a5daa3a0231a0fbba4f14db7ec463cf99d7768e
x64
```

### Extension Version

```console
ms-vscode.powershell@2023.8.0
```

### Steps to Reproduce

Execute the following code in VS Code:

# Set Backup Service credentials
$username = "VIAMONSTRA\SQL_Service"
$password = ConvertTo-SecureString "P@ssword" -AsPlainText -Force
$psCred = New-Object System.Management.Automation.PSCredential -ArgumentList ($username, $password)    
Set-Service -Name SMS_SITE_BACKUP -Credential $psCred

**Comment by SydneyhSmith** [comment]
Thanks for the issue @chrisecklar could you please provide the error message you are seeing along with visuals of what you are seeing?

**Comment by SydneyhSmith** [comment]
Could you please also share the output of $env:PSModulePath?

**Comment by chrisecklar** [comment]
Hello Sydney.

Here is the error I'm seeing as well as the results of the $env:PSModulePath command:
[cid:2b5febf3-c69c-4a3a-91d1-262ca126b573]


Chris Ecklar

Client Services – Admin | 1-3109

Desktop Support Technician – Senior

Deployment | MDM | Desktop Imaging

[cid:4d7f03d6-7659-47b0-8763-657798b5dfc7]Report all phishy email to: ***@***.******@***.***>[The History of McDonald's Filet-O-Fish | QSR magazine]

________________________________

**Comment by SydneyhSmith** [comment]
@chrisecklar it looks like your image didn't come through, can you please repost it? Thanks!

**Comment by chrisecklar** [comment]
Hello Sydney.

Here is the screenshot:
[cid:d2e0d241-8313-418d-a98b-dbe3627d82d3]


Chris Ecklar

Client Services – Admin | 1-3109

Desktop Support Technician – Senior

Deployment | MDM | Desktop Imaging

[cid:bab9be43-e02f-4150-ac3d-d4a97abb1e31]Report all phishy email to: ***@***.******@***.***>[The History of McDonald's Filet-O-Fish | QSR magazine]

________________________________

**Comment by andyleejordan** [comment]
@chrisecklar when you reply to an email with the picture, it appears your email client is removing it. We're not receiving it over here on GitHub, we just see:

> Here is the screenshot:
> [cid:d2e0d241-8313-418d-a98b-dbe3627d82d3]

Instead of a picture.

**Comment by chrisecklar** [comment]
Hello Andy.

Apologies for that—I'm not sure why that is happening to inline pictures.

I've attached the screenshot as an attachment this time.  Please let me know if you receive it.


Chris Ecklar

Client Services – Admin | 1-3109

Desktop Support Technician – Senior

Deployment | MDM | Desktop Imaging

[cid:e58cd5c9-6703-4a57-96ee-db4a291042f7]Report all phishy email to: ***@***.******@***.***>[The History of McDonald's Filet-O-Fish | QSR magazine]

________________________________

**Comment by andyleejordan** [comment]
Unfortunately no, I don't think that GitHub supports sending images in over email at all. You can access this issue in a browser at https://github.com/PowerShell/vscode-powershell/issues/4849 and upload it directly in a reply!

**Comment by chrisecklar** [comment]
Attached is the requested screenshot.
![thumbnail_thumbnail_image](https://github.com/PowerShell/vscode-powershell/assets/96747440/2c40b0fc-067c-4654-b816-20ea409cffa0)

**Comment by andyleejordan** [comment]
Ok @chrisecklar I think I have an answer for you: it appears that for the PowerShell Extension Terminal (used for debugging) you are in fact running Windows PowerShell 5.1, which does not support `-Credential`. Please see this [FAQ](https://github.com/PowerShell/vscode-powershell/blob/main/docs/troubleshooting.md#how-do-i-change-the-powershell-version) on how to change the version of PowerShell the extension itself is using to PowerShell 7.4.

**Comment by JustinGrote** [comment]
@chrisecklar can you confirm that switching to PS7 in the extension console fixed this for you?

**Comment by chrisecklar** [comment]
Hello Justin & Andy.  We can now mark this issue as resolved.  After returning from a long holiday break I completely rebuilt our POC environment, updated Visual Studio Code, PowerShell extensions and PwSH to the absolute latest versions and the issue went away.  I can now use the -Credential parameter (and set it) using the Visual Studio Code console.

I did not need to switch PowerShell extensions to PowerShell 5.x--I was able to leave it pointed to PwSH 7.4.x and the line of code worked as expected.

I'll keep your suggestions in mind though should I run into a similar issue in the future.

Happy New Year to you both, and thanks again for your assistance.
